pyomo/contrib/mindtpy/iterate.py

```diff
@@ -68,6 +68,15 @@ def MindtPy_iteration_loop(solve_data, config):
         else:
             raise NotImplementedError()
 
+        if config.add_regularization is not None:
```

Comment:
Regularization should only be considered if there is an existing incumbent (upper bound) solution. This would remove a part of the conditional in the next line.